plugins/modules/ec2_eip.py

```diff
@@ -234,13 +241,19 @@ class EipError(Exception):
 
 
 def associate_ip_and_device(
-    ec2, module, address, private_ip_address, device_id, allow_reassociation, check_mode, is_instance=True
-):
+    ec2,
+    module: AnsibleAWSModule,
+    address: Optional[Dict[str, str]],
+    private_ip_address: Optional[str],
+    device_id: str,
+    allow_reassociation: bool,
+    is_instance: bool = True,
+) -> bool:
     if address_is_associated_with_device(ec2, module, address, device_id, is_instance):
-        return {"changed": False}
+        return False
 
     # If we're in check mode, nothing else to do
-    if not check_mode:
+    if not module.check_mode:
```

Comment:
For readability, I would suggest maybe setting `address` to `{}` here if it's supplied as `None`, rather than having several `address or {}` checks below. Or better yet, don't make it optional and instead require the check outside this function. Same for a few other functions below.

Reply:
updated.